src/main/java/build/buf/protovalidate/internal/evaluator/MessageValue.java

```diff
@@ -18,33 +18,39 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * The {@link build.buf.protovalidate.internal.evaluator.Value} type that contains a {@link
  * com.google.protobuf.Message}.
  */
 public final class MessageValue implements Value {
 
-  /** Object type since the object type is inferred from the field descriptor. */
-  private final Object value;
+  private final ProtobufMessageLike value;
 
   /**
    * Constructs a {@link MessageValue} with the provided message value.
    *
    * @param value The message value.
    */
   public MessageValue(Message value) {
-    this.value = value;
+    this.value = new ProtobufMessageLike(value);
   }
 
   @Override
-  public Message messageValue() {
-    return (Message) value;
+  public MessageLike messageValue() {
+    return value;
   }
 
   @Override
-  public <T> T value(Class<T> clazz) {
-    return clazz.cast(value);
+  @Nullable
+  public <T> T jvmValue(Class<T> clazz) {
+    return null;
```

Comment:
Is this not used by anyone? Could we not keep it returning the underlying `Message` as before?

Do we need to make smaller interfaces and only implement `jvmValue` on `Value` types that provide a valid implementation?

Reply:
Correct, this is not used. `jvmValue` is used for fields that protovalidate-java validates directly itself, like enums (as opposed to passing into CEL), and right now it does not do any validation on message types.

See also the changed doc on `Value.jvmValue`.

Reply:
But even if it did, it doesn't make much sense to cast the message type to anything, does it? This hints at the narrower interface you mention.